docs/source/development/AlgorithmMPISupport.rst

```diff
@@ -0,0 +1,464 @@
+   :figwidth: 25%
+   :align: right
+
+   ``ExecutionMode::MasterOnly`` based on an input workspace with ``StorageMode::MasterOnly`` an no output. Example: ``Save`` or any other save algorithm.
+
+Just like the storage mode describes how data is stored, and execution mode describes how an algorithm is executed on this data.
+There are five execution modes (in namespace ``Parallel``):
+
+- ``ExecutionMode::Invalid`` is used to indicate that execution is not possible, e.g., if the storage modes of the inputs are inconsistent.
+- ``ExecutionMode::Serial`` is used for serial execution, i.e., in non-MPI builds of Mantid or if there is only a single MPI rank. Having this mode allows for running algorithms that do not support MPI in MPI builds by running only with a single MPI rank.
+- ``ExecutionMode::Identical`` is used for running an algorithm in an identical way on all MPI ranks. This would typically be used if the input workspaces have ``StorageMode::Cloned``.
+- ``ExecutionMode::Distributed`` is used for running an algorithm in a distributed way across all MPI ranks. This would typically be used if the input workspaces have ``StorageMode::Distributed``.
+- ``ExecutionMode::MasterOnly`` is typically used for running an algorithm if the input workspaces have ``StorageMode::MasterOnly``.
+
+The use of the word 'typically' above is intentional and indicates that there may be other cases.
+In particular, an algorithm may cause a transition from one storage mode to another, or may take inputs with different storage modes.
+Examples are given in the series of figures on the right.
+
+Building and Running Mantid with MPI Support
+############################################
+
+Build with MPI support
+----------------------
+
+To build Mantid with MPI support as described in this document run ``cmake`` with the additional option ``-DMPI_EXPERIMENTAL=ON``.
+This requires ``boost-mpi`` and a working MPI installation.
+
+Writing and running Python scripts
+----------------------------------
+
+In principle Python scripts that use only algorithms that support MPI can be run with MPI **without changes**.
+For example:
+
+.. code-block:: python
+
+  from mantid.simpleapi import *
+  
+  dataX = [1,2,3,4,2,3,4,5,3,4,5,6,4,5,6,7]
+  dataY = [1,1,1,1,1,1,1,1,1,1,1,1]
+  dataE = [1,1,1,1,1,1,1,1,1,1,1,1]
+
+  # CreateWorkspace has a new property called ParallelStorageMode that allows setting the
+  # desired storage mode. It defaults to "Parallel::StorageMode::Cloned".
+  dataWS = CreateWorkspace(DataX=dataX, DataY=dataY, DataE=dataE, NSpec=4, UnitX="Wavelength", ParallelStorageMode="Parallel::StorageMode::Distributed")
+  ws = Rebin(dataWS, "1,1,7");
+
+  print("Histograms: " + str(ws.getNumberHistograms()))
+  for i in range(ws.getNumberHistograms()):
+      print("(Local) workspace index: " + str(i))
+      print(ws.readX(i))
+      print(ws.readY(i))
+
+
+Run Python with ``mpirun`` and the desired number of MPI ranks:
```

Comment:
This could be reworded I think. You first say that it can be run with python given the commands and then go on to say that it's not possible to run the commands. Maybe we need to explain what else is required.